server.c

```
@@ -84,6 +85,22 @@ static int static_dir_len = -1;
 static khash_t(mime_type)* mime_type;
+typedef struct {
+  char* path;
+  char* body;
+  size_t body_len;
+  const char* ctype;
+  char* header_keep_alive;
+  size_t header_keep_alive_len;
+  char* header_close;
+  size_t header_close_len;
+} file_cache_entry;
+KHASH_MAP_INIT_STR(file_cache, file_cache_entry*)
```

Clang (14.0.6)

[warning] function 'kh_put_file_cache' has cognitive complexity of 29 (threshold 25)

(readability-function-cognitive-complexity)